src/conn/conn_api.c

```diff
@@ -663,6 +663,100 @@ __wt_conn_remove_extractor(WT_SESSION_IMPL *session)
     return (ret);
 }
 
+/*
+ * __tiered_confchk --
+ *     Check for a valid tiered storage source.
+ */
+static int
+__tiered_confchk(
+  WT_SESSION_IMPL *session, WT_CONFIG_ITEM *cname, WT_NAMED_STORAGE_SOURCE **nstoragep)
+{
+    WT_CONNECTION_IMPL *conn;
+    WT_NAMED_STORAGE_SOURCE *nstorage;
+
+    *nstoragep = NULL;
+
+    if (cname->len == 0 || WT_STRING_MATCH("none", cname->str, cname->len))
+        return (0);
+
+    conn = S2C(session);
+    TAILQ_FOREACH (nstorage, &conn->storagesrcqh, q)
+        if (WT_STRING_MATCH(nstorage->name, cname->str, cname->len)) {
+            *nstoragep = nstorage;
+            return (0);
+        }
+    WT_RET_MSG(session, EINVAL, "unknown storage source '%.*s'", (int)cname->len, cname->str);
+}
+
+/*
+ * __wt_tiered_bucket_config --
+ *     Given a configuration, configure the bucket storage.
+ */
+int
+__wt_tiered_bucket_config(WT_SESSION_IMPL *session, WT_CONFIG_ITEM *cval, WT_CONFIG_ITEM *bucket,
+  WT_BUCKET_STORAGE **bstoragep)
+{
+    WT_BUCKET_STORAGE *bstorage, *new;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_NAMED_STORAGE_SOURCE *nstorage;
+    uint64_t hash_bucket, hash;
+
+    *bstoragep = NULL;
+
+    bstorage = new = NULL;
+    conn = S2C(session);
+
+    __wt_spin_lock(session, &conn->storage_lock);
+
```

Comment:
I think it will be minimal, just like encryption. We have to protect against two threads creating simultaneously. We have a  spinlock for encryption and that spinlock has not been any problem. We do not collect stats within `wt_spin_lock`.